jlm/tooling/CommandLine.hpp

```diff
@@ -174,33 +175,33 @@ class JlmOptCommandLineOptions final : public CommandLineOptions
     inline static const char * NodeReduction_ = "NodeReduction";
   };
 
-  struct StatisticsCommandLineArgument
-  {
-    inline static const char * Aggregation_ = "print-aggregation-time";
-    inline static const char * AgnosticMemoryNodeProvisioning_ =
-        "print-agnostic-memory-node-provisioning";
-    inline static const char * AndersenAnalysis_ = "print-andersen-analysis";
-    inline static const char * Annotation_ = "print-annotation-time";
-    inline static const char * CommonNodeElimination_ = "print-cne-stat";
-    inline static const char * ControlFlowRecovery_ = "print-cfr-time";
-    inline static const char * DataNodeToDelta_ = "printDataNodeToDelta";
-    inline static const char * DeadNodeElimination_ = "print-dne-stat";
-    inline static const char * FunctionInlining_ = "print-iln-stat";
-    inline static const char * InvariantValueRedirection_ = "printInvariantValueRedirection";
-    inline static const char * JlmToRvsdgConversion_ = "print-jlm-rvsdg-conversion";
-    inline static const char * LoopUnrolling_ = "print-unroll-stat";
-    inline static const char * MemoryStateEncoder_ = "print-basicencoder-encoding";
-    inline static const char * PullNodes_ = "print-pull-stat";
-    inline static const char * PushNodes_ = "print-push-stat";
-    inline static const char * ReduceNodes_ = "print-reduction-stat";
-    inline static const char * RegionAwareMemoryNodeProvisioning_ =
-        "print-memory-node-provisioning";
-    inline static const char * RvsdgConstruction_ = "print-rvsdg-construction";
-    inline static const char * RvsdgDestruction_ = "print-rvsdg-destruction";
-    inline static const char * RvsdgOptimization_ = "print-rvsdg-optimization";
-    inline static const char * SteensgaardAnalysis_ = "print-steensgaard-analysis";
-    inline static const char * ThetaGammaInversion_ = "print-ivt-stat";
-  };
+  static inline const util::BijectiveMap<util::Statistics::Id, std::string>
```

Comment:
Declaring this in the header with `inline` might add the construction of this object to the static initialization of all translation units that include `CommandLine.hpp`. Making it just static, and moving the initialization to `CommandLine.cpp` could work. I will test if it at all makes a difference.

Reply:
Or at the very least we could use `std::string_view` instead of `std::string`, since these strings are constants with static lifetime, to avoid copying into heap allocated strings all over the place.

Reply:
Great catch! I changed it, please have a look again.